File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    answer = []
    change_count = 0
    zero_count = 0
    
    while s != "1":
        tmp = 0
        for i in s:
            if i == "1":
                tmp+=1
        
        zero_count += len(s) - tmp
        s=str(bin(tmp)[2:])
        change_count+=1
    answer = [change_count,zero_count]
    return answer

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {
        int[] answer = new int[2];
        
        int zeroCount = 0;
        int count = 0;
        while(!s.equals("1")){
            
            count++;
            
            int originLength = s.length();
            s = s.replace("0","");
            int changeLength = s.length();
            
            zeroCount += originLength - changeLength;
            
            s = Integer.toBinaryString(changeLength);
            
        }
        
        answer[0] = count;
        answer[1] = zeroCount;
        return answer;
    }
    
    
}